### PR TITLE
CI cleanup

### PR DESCRIPTION
**Pull request by andersy005**
- Removes duplicated environment files
- Adds dependabot configuration for keeping GitHub actions up-to-date
- Updates pre-commit hooks
- Disable GitHub workflows on forks

- [x] Passes `pre-commit run --all-files`

### CODE REVIEW

File: doc/.gitignore [hunk 1]
```diff
@@ -1 +1 @@
-xnemogcm*
\ No newline at end of file
+xnemogcm*
```
Comment (jbusecke) [inline doc/.gitignore:1]: Just curious where this is from?

File: doc/contributor_guide.rst [hunk 2]
```diff
@@ -23,18 +23,18 @@ The easiest way to start developing xgcm pull requests,
 is to install one of the conda environments provided in the `ci folder <https://github.com/xgcm/xgcm/tree/master/ci>`_::
 
     conda env create -f ci/environment-py3.8.yml
-
```
Comment (jbusecke) [inline doc/contributor_guide.rst:26]: Are you using some sort of rst linter? Should that be part of the CI?

File: doc/images/axis_positions.svg [hunk 3]
```diff
@@ -56,7 +56,7 @@ fill: #000000;
 <path d="M0,0
 L0,230
 L650,230
-L650,0 Z " 
+L650,0 Z "
```
Comment (jbusecke) [inline doc/images/axis_positions.svg:59]: A bit unsure about all of these changes? Was this some linter aswell?

File: xgcm/test/test_transform.py [hunk 4]
```diff
@@ -812,7 +812,7 @@ def test_conservative_interp_warn():
 
 @pytest.mark.skipif(numba is None, reason="numba required")
 def test_grid_transform_noname_data(multidim_cases):
-    """ Check handling of a `da` input without name"""
```
Comment (jbusecke) [inline xgcm/test/test_transform.py:815]: I guess this is just due to an updated black version?

File: .github/workflows/ci.yaml [hunk 5]
```diff
@@ -33,15 +38,13 @@ jobs:
           activate-environment: test_env_xgcm # Defined in ci/environment*.yml
           auto-update-conda: false
           python-version: ${{ matrix.python-version }}
```
Comment (jbusecke) [inline .github/workflows/ci.yaml:40]: Just trying to understand this more. Is this line here installing a specific python version? Or is that set somewhere else?

File: .github/workflows/linting.yaml [hunk 6]
```diff
@@ -6,14 +6,15 @@ on:
...
+      name: pre-commit
+      if: github.repository == 'xgcm/xgcm'
+      runs-on: ubuntu-latest
+      steps:
+          - name: checkout
+            uses: actions/checkout@v2
+          - name: set up python
+            uses: actions/setup-python@v2
+            with:
+                python-version: 3.8
```
Comment (jbusecke) [inline .github/workflows/linting.yaml:18]: Should we update this to 3.9?

**Review by jbusecke — APPROVED** [review]
Looks great, thanks @andersy005. I basically just have questions to understand this more.

File: doc/.gitignore (hunk 1 above)
Comment (andersy005) [reply doc/.gitignore:1]: Hmmm... I don't know how this sneaked in (I don't recall doing anything with the `.gitignore` file)

File: doc/images/axis_positions.svg (hunk 3 above)
Comment (andersy005) [reply doc/images/axis_positions.svg:59]: I think these white space removals were introduced by the new pre-commit hooks I added:

```yaml
  - repo: https://github.com/pre-commit/pre-commit-hooks
    rev: v4.0.1
    hooks:
      - id: trailing-whitespace
      - id: end-of-file-fixer
      - id: check-yaml
```

Let me know if you want us to revert back to the original versions.

File: xgcm/test/test_transform.py (hunk 4 above)
Comment (andersy005) [reply xgcm/test/test_transform.py:815]: Yes. I updated the pre-commit hook version

File: doc/.gitignore (hunk 1 above)
Comment (rcaneill) [reply doc/.gitignore:1]: This came from when I created the NEMO example ( PR #192 ).

File: doc/images/axis_positions.svg (hunk 3 above)
Comment (jbusecke) [reply doc/images/axis_positions.svg:59]: Nope, I like all of this. Just want to understand it too hehe

**Pull request merged**